_docs/rfc/imports.md

```diff
@@ -0,0 +1,963 @@
+module "mysql" {
+  # args omitted for brevity
+  vpc_id = module.vpc.outputs.vpc_id
+}
+```
+
+This example reuses the outputs of `module.vpc` across the two modules, which is the equivalent of having the `vpc`
+`dependency` block redefined in the two module configs.
+
+**Keeping remote state configuration DRY**
+
+This example is covered in [the original issue that proposed this
+idea](https://github.com/gruntwork-io/terragrunt/issues/759).
+
+
+#### hcldecode or read_terragrunt_config helper function
+
+Instead of defining a dedicated block for `import`, we could define a helper function that parses the relevant config.
+For example, the explicit triple import example in the [hierarchical variables use
+case](#hierarchical-variables-included-across-multiple-terragrunt-hcl-files) can be implemented as:
+
+```
+locals {
+  root_config = read_terragrunt_config("../../../root.hcl")
+  region_config = read_terragrunt_config("../../region.hcl")
+  env_config = read_terragrunt_config("../env.hcl")
+}
+
+inputs = merge(
+  local.root_config.inputs,
+  local.region_config.inputs,
+  local.env_config.inputs,
+  {
+    # args to module
+  },
+)
+```
+
+Pros:
+
+- Relatively simple implementation. It is significantly easier to add helper functions than it is to introduce new
+  blocks in terragrunt.
+- Can address all the same use cases as `import` blocks.
+
+Cons:
+
+- Everything has to be explicit in the config. We can not support automatic merging capabilities when using helper
+  functions, as you can't manipulate the config in a helper. This can lead to very verbose configurations if one wants
+  to use this in place of `include`.
+
+Note that to take full advantage of this approach, all the blocks in `terragrunt.hcl` need to be redefined as attributes
+so that we can use assignment to override them.
+
+Let's walk through a few more of the use cases:
+
+**Keeping remote state configuration DRY**
+
+parent
+
+```hcl
+remote_state {
+  backend = "s3"
+  config = {
+    bucket         = "my-terraform-state"
+    key            = "${path_relative_to_include()}/terraform.tfstate"
+    region         = "us-east-1"
+    encrypt        = true
+    dynamodb_table = "my-lock-table"
+  }
+}
+```
+
+child
+
+```hcl
+locals {
+  root_config = read_terragrunt_config("../../../root.hcl")
+}
+
+remote_state {
+  backend = local.root_config.remote_state.backend
+  config = merge(
+    local.root_config.remote_state.config,
+    {
+      # The relative path between the root terragrunt directory and the current terragrunt file directory.
+      key =  relpath(import.root.terragrunt_dir, get_terragrunt_dir())
+    },
+  )
+}
+```
+
+Note that we can't do:
+
+```hcl
+remote_state = deep_merge(local.root_config.remote_state, { config = { key = relpath } })
+```
+
+due to the fact that `remote_state` is a block and not an attribute.
+
+
+**Reusing dependencies**
+
+We can't reuse `dependency` blocks in this implementation because there is no way to auto merge the blocks.
```

Comment:
I'm not sure we'd need `dependency` blocks at all... Each account is standalone and dependencies between modules within an account are already captured.

Reply:
This is actually the section for read_terragrunt_config, not the approach for single tg file. See line 709

Reply:
Doh!